File: web/docs-examples/python/examples/chat_basic.py
"""Basic chat completion with OpenAI.

This example loads a .prompty file and runs a simple chat completion.
Used in: how-to/openai.mdx, getting-started/index.mdx
"""
from __future__ import annotations

from prompty import invoke, load

agent = load("chat-basic.prompty")
result = invoke(agent, inputs={"question": "What is Prompty?"})
print(result)

File: web/docs-examples/typescript/examples/chat-basic.ts
/**
 * Basic chat completion — load a .prompty file and invoke it.
 *
 * @example
 * ```bash
 * OPENAI_API_KEY=sk-... npx tsx examples/chat-basic.ts
 * ```
 */
import "@prompty/openai"; // auto-registers openai executor + processor
import { invoke } from "@prompty/core";
import { resolve } from "node:path";

const promptyFile = resolve(import.meta.dirname, "../../prompts/chat-basic.prompty");

export async function chatBasic(question?: string): Promise<string> {
  const result = await invoke(promptyFile, {
    question: question ?? "What is Prompty?",
  });
  return result as string;
}

// Run directly
const response = await chatBasic();
console.log(response);
